overlord/servicestate/servicestate.go

```diff
@@ -119,3 +122,76 @@ func Control(st *state.State, appInfos []*snap.AppInfo, inst *Instruction, conte
 
 	return tts, nil
 }
+
+// StatusDecorator supports decorating client.AppInfos with service status.
+type StatusDecorator struct {
+	sysd systemd.Systemd
+}
+
+// NewStatusDecorator returns a new StatusDecorator.
+func NewStatusDecorator(rep interface {
+	Notify(string)
+}) *StatusDecorator {
+	return &StatusDecorator{
+		sysd: systemd.New(dirs.GlobalRootDir, systemd.SystemMode, rep),
+	}
+}
+
+// DecorateWithStatus adds service status information to the given client.AppInfo associated with the given snap.AppInfo.
+func (sd *StatusDecorator) DecorateWithStatus(appInfo *client.AppInfo, snapApp *snap.AppInfo) error {
+	if appInfo.Snap != snapApp.Snap.InstanceName() || appInfo.Name != snapApp.Name {
+		return fmt.Errorf("misassociated app info %v and client app info %s.%s", snapApp, appInfo.Snap, appInfo.Name)
+	}
+	if !snapApp.IsService() {
+		// nothing to do
+		return nil
+	}
+
+	// collect all services for a single call to systemctl
+	serviceNames := make([]string, 0, 1)
```

Comment:
Not a problem by any means, but wondering why the change from the original code:
`serviceNames := make([]string, 0, 1+len(app.Sockets)+1)` ?

Reply:
no, you are right, this got lost because I re-enabled part of this code piece by piece

Reply:
did something along those lines